### PR TITLE
Implement mutationID generation

### PR DESCRIPTION
**Pull request by aboodman**
Toward #94

### CODE REVIEW

File: src/db/commit.rs [hunk 1]
```diff
@@ -448,29 +460,31 @@ mod tests {
...
-                assert_eq!(sm.server_state_id(), "server_state_id");
+                assert_eq!(sm.last_mutation_id(), 2);
+                assert_eq!(sm.server_state_id(), "server_state_id 2");
             }
         }
-        assert_eq!(local.meta().local_create_date(), "local_create_date");
-        assert_eq!(local.meta().basis_hash(), Some("basis_hash"));
-        assert_eq!(local.meta().checksum(), "checksum");
-        assert_eq!(local.value_hash(), "value_hash");
+        assert_eq!(snapshot.meta().local_create_date(), "local_create_date 2");
```
Comment (aboodman) [inline src/db/commit.rs:483]: whoops

File: src/db/commit.rs [hunk 2]
```diff
@@ -105,6 +105,18 @@ impl Commit {
         self.commit().value_hash().unwrap()
     }
 
+    pub fn next_mutation_id(&self) -> u64 {
+        let meta = self.commit().meta().unwrap();
```
Comment (phritz) [inline src/db/commit.rs:109]: Not something that came in with this PR but: the highly overloaded nature of "commit" in this code I found super confusing. commit()'s signature is basically the following, which is hard to make heads or tails of:
```
struct Foo {}

impl Foo {
  fn foo() -> foo::Foo
}
```
Can we bring `super::commit_generated::commit` in as `commit_fb` or similar and call `commit()` something like `fb()` so the reader can see what is what?

File: src/db/commit.rs [hunk 3]
```diff
@@ -105,6 +105,18 @@ impl Commit {
...
+        let mid = match meta.typed_type() {
+            commit::MetaTyped::LocalMeta => meta.typed_as_local_meta().unwrap().mutation_id(),
```
Comment (phritz) [inline src/db/commit.rs:111]: Do you have creating a genesis snapshot when we create the db on your list of todos?

File: src/db/commit.rs (hunk 2 above)
Comment (aboodman) [reply src/db/commit.rs:109]: That's a good idea. https://github.com/rocicorp/repc/issues/100

File: src/db/commit.rs (hunk 3 above)
Comment (aboodman) [reply src/db/commit.rs:111]: Nope! https://github.com/rocicorp/repc/issues/102

**Pull request merged**